infrastructure-petstore-rest-client/src/test/java/today/stepbeyond/examples/springbootexamples/infrastructure/gateways/JsonDifferenceAssertion.java

```diff
@@ -0,0 +1,46 @@
+package today.stepbeyond.examples.springbootexamples.infrastructure.gateways;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.HashMap;
+import java.util.Map;
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.Assertions;
+import wiremock.com.google.common.collect.MapDifference;
+import wiremock.com.google.common.collect.Maps;
+
+public class JsonDifferenceAssertion extends AbstractAssert<JsonDifferenceAssertion, String> {
+
+  protected JsonDifferenceAssertion(String s) {
+    super(s, JsonDifferenceAssertion.class);
+  }
+
+  public static JsonDifferenceAssertion assertThat(String actual) {
+    return new JsonDifferenceAssertion(actual);
+  }
+
+  public void isSameAs(String expected) {
+
+    var objectMapper = new ObjectMapper();
+    var type = new TypeReference<HashMap<String, Object>>() {};
+
+    try {
+      Map<String, Object> actualMap = objectMapper.readValue(actual, type);
+      Map<String, Object> expectedMap = objectMapper.readValue(expected, type);
+
+      MapDifference<String, Object> difference = Maps.difference(actualMap, expectedMap);
+
+      Assertions.assertThat(difference.areEqual())
```

Comment:
Nit: i think i would rather split this assertion block into seperate assertions. That would make this code more verbose, but could result in a more usable assertion error message. Maybe something like this: 

`// Assertion for missing objects
Assertions.assertThat(difference.entriesOnlyOnLeft())
    .as("Unexpected objects found in actual map: %s", difference.entriesOnlyOnLeft())
    .isEmpty();
Assertions.assertThat(difference.entriesOnlyOnRight())
    .as("Expected objects missing in actual map: %s", difference.entriesOnlyOnRight())
    .isEmpty();
`
But i am not 100% sure if that code snipped does the same :D

Reply:
Like that! In addition, these assertions can be wrapped with soft assertions.